closed/src/java.base/share/classes/com/sun/crypto/provider/NativeChaCha20Cipher.java

```diff
@@ -982,15 +1015,16 @@ public synchronized int doUpdate(byte[] in, int inOff, int inLen, byte[] out,
                     throw new ShortBufferException("Output buffer too small");
                 }
 
-                Objects.checkFromIndexSize(inOff, inLen, in.length);
+                if (in != null) {
+                    Objects.checkFromIndexSize(inOff, inLen, in.length);
 
-                byte aadArray[] = aadBuf.toByteArray();
-                aadBuf.reset();
-                int ret = nativeCrypto.ChaCha20Update(context, in, inOff, inLen, out, outOff, aadArray, aadArray.length);
-                if (ret == -1) {
-                    throw new ProviderException("Error in Native CipherBlockChaining");
+                    byte aadArray[] = aadBuf.toByteArray();
```

Comment:
Please use a more conventional declaration (`[]` in the type):
```java
                    byte[] aadArray = aadBuf.toByteArray();
```
Likewise on line 1141.

Reply:
Changed.